### PR TITLE
PR to find a proportional cake allocation

### PR DESCRIPTION
**Pull request by EranK123**
Hello, I am a student at Ariel University under Dr Erel Segal.(https://github.com/erelsgl)
As a part of Research Algorithms course I have implemented an algorithm for finding a proportional cake allocation in a almost bridgeless graph. The Implementation is based on the article: (https://arxiv.org/abs/1910.14129).

### CODE REVIEW

**Comment by EranK123** [comment]
Done

File: networkx/algorithms/approximation/contiguous_oriented_labeling.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
+"""
+This file admits a contiguous oriented labeling in an almost bridgeless graph.
```
Comment (erelsgl) [inline networkx/algorithms/approximation/contiguous_oriented_labeling.py:2]: The title should be much more detailed. It should contain a description of what is contiguous oriented labeling, what is almost bridgeless, what paper the algorithm comes from (including title, authors and link), and who the programmer is.

File: networkx/algorithms/approximation/contiguous_oriented_labeling.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+"""
+import math
+
+import matplotlib.pyplot as plt
+
+import networkx as nx
+import networkx.algorithms.isomorphism as iso
+
+
+def graph_example1():
```
Comment (erelsgl) [inline networkx/algorithms/approximation/contiguous_oriented_labeling.py:12]: All examples used in the doctest should be inside the doctest.

File: networkx/algorithms/approximation/contiguous_oriented_labeling.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    the vertex labeled i+ belongs to one of these edges.
+
+     References
+    ----------
+    Based on an article by Xiaohui Bei and Warut Suksompong
+    https://arxiv.org/pdf/1910.14129.pdf#subsection.4.1 - 2019
+
+    Programmer : Eran Katz
+
+    >>> get_contiguous_oriented_labeling(graph_example1())
```
[the remark this answers is not in the document]
Comment (erelsgl) [reply networkx/algorithms/approximation/contiguous_oriented_labeling.py:93]: Also, it is not clear why the "InEdgeView" is in the expected output. The expected output should contain only what is needed to verify the algorithm.

File: networkx/algorithms/approximation/contiguous_oriented_labeling.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+                oriented_graph.add_edge(x, y)
+                t += 1
+                t_update = False
+
+    return ordered_edges, oriented_graph.in_edges
+
+
+def almost_bridgeless_to_bridgless(g):
+    """
+    This function takes an almost bridgeless graph and adds an edge so it makes it bridgeless
```
Comment (erelsgl) [inline networkx/algorithms/approximation/contiguous_oriented_labeling.py:234]: Add explanations and examples. See here: https://github.com/networkx/networkx/pull/5585/files  how good examples look like.

File: networkx/algorithms/approximation/contiguous_oriented_labeling.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                nodes_in_ears.append(path[i])
+            if path[i + 1] not in nodes_in_ears:
+                nodes_in_ears.append(path[i + 1])
+    idx = 0
+    x_idx = math.inf
+    t = x_idx
+    c = 0
+    cycle = False  # a flag to determine if an ear is a cycle. if x = z
+    t_update = True  # a flag to determine where to insert edges in order
+    for (
```
Comment (erelsgl) [inline networkx/algorithms/approximation/contiguous_oriented_labeling.py:122]: This should be a single line: 
```
for edge in graph edges:
```

File: examples/algorithms/proportional_cake_allocation.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+This file gets a proportional allocation in a graph for 2 agents.
+"""
+import PiecewiseConstantValuation as pc
+import networkx as nx
+from networkx.algorithms.approximation import contiguous_oriented_labeling as col
+import matplotlib.pyplot as plt
+import networkx.algorithms.isomorphism as iso
+
+
+def graph_example1():
```
Comment (erelsgl) [inline examples/algorithms/proportional_cake_allocation.py:11]: Functions used only in doctests should be defined in the doctest

File: examples/algorithms/proportional_cake_allocation.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    60
+    """
+    s = 0
+    for i in v:
+        s += i.total_value()
+    return s
+
+
+def get_valuations_until_i(ordered_edges, vals_to_edges, edge, half):
+    """
```
Comment (erelsgl) [inline examples/algorithms/proportional_cake_allocation.py:116]: doctest is missing

**Review by erelsgl — CHANGES_REQUESTED** [review]

**Review by rossbar — COMMENTED** [review]
This has been stale for a while so I will go ahead and close it.

**Pull request closed without merging**